### PR TITLE
Improve search-time benchmarking with the nested track.

### PR DESCRIPTION
**Pull request by jpountz**
This commit adds benchmarks for nested sorting, nested aggregations as well
as queries that do not use the nested structure. The latter might sound
surprising but when nested mappings are used, Elasticsearch needs to add
filters in order to make sure that queries may only match root documents, so
it is important to verify we make these filters efficient.

It also changes the existing nested query so that it matches more documents
in order to exercise the join logic more. The previous queries were very
selective and returned in a couple milliseconds.

### CODE REVIEW

File: nested/track.json [hunk 1]
```diff
@@ -20,17 +20,59 @@
...
 		"name": "randomized-nested-queries",
 		"operation-type": "search",
 		"param-source": "nested-query-source",
 		"cache": false
-	}],
+        }, {
+                "name": "randomized-term-queries",
+                "operation-type": "search",
+                "param-source": "term-query-source",
+                "cache": false
```
Comment (danielmitterdorfer) [inline nested/track.json]: Nit: If you call this `use_request_cache` and pass this value with `self._params["use_request_cache"]` in the custom parameter source, it's better aligned with Rally core.

File: nested/track.py [hunk 2]
```diff
@@ -18,15 +16,59 @@ def __init__(self, indices, params):
...
+                            "mode" :  "max",
+                            "order": "desc",
+                            "nested_path": "answers"
+                        }
+                    }
+                ]
+            },
+            "index": None,
+            "type": None,
+            "use_request_cache": False
```
Comment (danielmitterdorfer) [inline nested/track.py]: Here you can do: `"use_request_cache": self._params["use_request_cache"]`. This makes it better aligned with Rally core.

File: nested/track.json [hunk 3]
```diff
@@ -20,17 +20,59 @@
...
+        }, {
+                "name": "randomized-sorted-term-queries",
+                "operation-type": "search",
+                "param-source": "sorted-term-query-source",
+                "cache": false
```
Comment (danielmitterdorfer) [inline nested/track.json]: same as above w.r.t. to `use_request_cache`.

File: nested/track.json [hunk 4]
```diff
@@ -20,17 +20,59 @@
...
+	}, {
+                "name": "match-all",
+                "operation-type": "search",
+                "body": {
+                        "query": {
+                                "match_all": {}
+                        }
+                },
+                "cache": false
```
Comment (danielmitterdorfer) [inline nested/track.json]: same as above w.r.t. to `use_request_cache`

File: nested/track.json [hunk 5]
```diff
@@ -20,17 +20,59 @@
...
+                                                        "date_histogram": {
+                                                                "field": "answers.date",
+                                                                "interval": "month"
+                                                        }
+                                                }
+                                        }
+                                }
+                        }
+                },
+                "cache": false
```
Comment (danielmitterdorfer) [inline nested/track.json]: same as above w.r.t. to `use_request_cache`

File: nested/track.py [hunk 6]
```diff
@@ -18,15 +16,59 @@ def __init__(self, indices, params):
...
+            "body": {
+                "query": {
+                    "match": {
+                        "tag": "%s" % random.choice(self.tags)
+                    }
+                }
+            },
+            "index": None,
+            "type": None,
+            "use_request_cache": False
```
Comment (danielmitterdorfer) [inline nested/track.py]: same as above w.r.t. to `use_request_cache`

File: nested/track.json (hunk 1 above)
Comment (jpountz) [inline nested/track.json]: maybe we should update `docs/adding_tracks.rst`?

**Comment by jpountz** [comment]
@danielmitterdorfer I pushed a new commit, is it what you had in mind?

**Pull request merged**